src/main/java/com/quorum/gauge/services/TransactionService.java

```diff
@@ -329,7 +332,10 @@ public TransactionReceipt waitForTransactionReceipt(QuorumNode node, String tran
             throw new RuntimeException(e);
         }
 
-        String data = c.getContractBinary();
+
+        //TODO change 0 by initVal
+        String encodedConstructor = FunctionEncoder.encodeConstructor(Arrays.<Type>asList(new org.web3j.abi.datatypes.generated.Uint256(0)));
```

Comment:
that is strange. do we know the reason?
do we need to increase `web3j` version so the class `Contract` handles this case on the `getContractBinary()` method?

Reply:
That was my intuition as well, but the getContractBinary() might be used as is (just the smart contract, without the `initVal`).
I need to get the reference to that call to see if we could just add the initVal or not (and I don't know yet if the initVal is part of the Contract class, have to check).

Reply:
It seems the contract doesn't have an `initVal` field, so we should pass it as an arg to `getContractBinary`.
In the acceptance test, it's just used there for gas estimate. But in the wild, I don't know how much it is used and if it would break something.

Reply:
👍

Reply:
lets add in the TODO or leave a comment to a new issue where we should fix this in the future

Reply:
The test doesn't take any specific initValue. Should we update the spec to have a parametrizable initVal that we would pass along? Or 0 is good enough?

Reply:
I think for this case, 0 is good enough.